torch/_logging/_internal.py

```diff
@@ -831,7 +831,7 @@ def _init_logs(log_file_name=None):
 
         if log_file_name is not None:
             _setup_handlers(
-                lambda: logging.FileHandler(log_file_name),
+                lambda: logging.FileHandler(log_file_name, 'w+'),
```

Comment:
I would agree with Geeta that you most likely want to have both options.
I don't know what are the precedents for this but I could see `TORCH_LOGS_OUT=foo.txt` to override and `TORCH_LOGS_OUT=+foo.txt` to appends. Using the fact that a path starting with + is not valid.

Reply:
Oh I like this idea - happy to switch to that (@oulgen does that sound good to you?)

Reply:
Sounds good. Feel free to implement it, otherwise I can do it when I’m back from pto

Reply:
will do in this PR!